### ISSUE TITLE
Posts: default posts that come with layout are timestamped in UTC but user posts are in local timezone

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#49221]
### Steps to reproduce the behavior

Prereq. user timezone is not UTC+0.

1. Create a new site.
2. Select any layout. I used `Stratford`.
3. Once site is created, create another post.
4. Go to My Sites > Posts.
5. Open all three posts (2 pre-generated posts and your post from Step 3).
6. On the right sidebar, click on the Posts tab.
7. Check the published date for all three posts.

#### What I expected to happen
Published dates for all posts should be aligned with user's timezone.

#### What actually happened
Default/pre-generated posts that come with the layout are set to UTC+0.
User-generated posts are set to whatever timezone is set for the user.

### Context
User scenario manual testing.

#### Browser / OS version
Browser: Firefox Nightly 86.0a1
OS: macOS Catalina 10.15.7

#### Is this specific to the applied theme? Which one?
Should happen on all themes.

#### Does this happen on simple or atomic sites or both?
Tested on simple.

#### Is there any console output or error text?
No.

#### Level of impact (Does it block purchases? Does it affect more than just one site?)
Moderate - this causes the `Latest posts` block to show unexpected ordering, as well as confuse users that live hours behind UTC+0.

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.

https://user-images.githubusercontent.com/6549265/105560403-7de05300-5cc8-11eb-8118-2a6663fada57.mov

![Screen Shot 2021-01-22 at 15 36 02](https://user-images.githubusercontent.com/6549265/105560418-859ff780-5cc8-11eb-9366-67fa9732699b.png)
![Screen Shot 2021-01-22 at 15 35 53](https://user-images.githubusercontent.com/6549265/105560422-889ae800-5cc8-11eb-9590-fce9b6b73a27.png)

**Comment by obenland** [comment]
This sounds like it's an issue with default content on new sites? @simison Is this something you're familiar with?

**Comment by simison** [comment]
Yeah, @ianstewart and I were considering to not add this demo-content at all since removing/editing it is confusing, which would make this problem to go away as well.

**Comment by jordesign** [comment]
@simison @ianstewart  Keen for your opinion if this is a bug worth addressing as part of the Blogger Experience Pod?
